experiment/adscert/signer.go

```diff
@@ -0,0 +1,108 @@
+package adscert
+
+import (
+	crypto_rand "crypto/rand"
+	"encoding/base64"
+	"encoding/pem"
+	"fmt"
+	"github.com/IABTechLab/adscert/pkg/adscert/api"
+	"github.com/IABTechLab/adscert/pkg/adscert/discovery"
+	"github.com/IABTechLab/adscert/pkg/adscert/signatory"
+	"github.com/benbjohnson/clock"
+	"github.com/prebid/prebid-server/config"
+	"time"
+)
+
+const SignHeader = "X-Ads-Cert-Auth"
+
+type Signer interface {
+	Sign(destinationURL string, body []byte) (string, error)
+}
+
+func NewAdCertsSigner(experimentAdCertsConfig config.ExperimentAdCerts) Signer {
+	if !experimentAdCertsConfig.Enabled {
+		return nil
```

Comment:
I recommend to create and return a NilSigner instead of a nil value, a la https://sourcemaking.com/design_patterns/null_object.

Reply:
Yes, so much better with this change now.